client/src/pages/Health.jsx

```diff
@@ -0,0 +1,182 @@
+  const calculateBmi = async () => {
+    if (!validateInputs()) return;
+
+    setIsLoading(true);
+    await new Promise(resolve => setTimeout(resolve, 500));
+
+    const weightValue = parseFloat(weight);
+    const heightValue = parseFloat(height);
+
+    if (weightValue && heightValue) {
+      const bmiValue = (weightValue / (heightValue * heightValue)).toFixed(2);
+      setBmi(bmiValue);
+      setMessage(`BMI successfully calculated! Your BMI is ${bmiValue}.`);
+      updateMilestones(bmiValue);
+      saveBmiToBackend(bmiValue);  // Save BMI to backend after calculation
+    } else {
+      setMessage("Invalid weight or height.");
+    }
+
+    setIsLoading(false);
+  };
+
+  // Function to update milestones including the date formatted next to the checkmark
+  const updateMilestones = (bmiValue) => {
+    const currentDate = new Date().toLocaleDateString(); // Get the current date
+    let updates = [];
+
+    if (bmiValue < 18.5) {
+      updates.push(`✅ ${currentDate} - BMI is below the healthy range. Consider gaining weight.`);
+    } else if (bmiValue >= 18.5 && bmiValue < 25) {
+      if (bmiValue <= targetBmi) {
+        updates.push(`✅ ${currentDate} - Achieved target BMI of ${targetBmi}.`);
+      } else {
+        updates.push(`✅ ${currentDate} - Reached a healthy BMI range (18.5 - 24.9).`);
+      }
+    } else if (bmiValue > 24.9) {
+      updates.push(`✅ ${currentDate} - BMI is above the healthy range. Consider losing weight.`);
+    }
+
+    setMilestones(updates);  // Directly replace the milestones with the updated list
+  };
+
+  // Function to save BMI data to the backend
+  const saveBmiToBackend = async (bmiValue) => {
+    console.log(currentUser)
+    try {
+      const response = await fetch("/api/bmi/", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "x-csrf-token": csrfToken,  // Include CSRF token in headers
+        },
+        body: JSON.stringify({
+          userId: currentUser._id,  // Replace with the logged-in user's ID
+          bmi: bmiValue,
+          note: "Auto-generated BMI entry"
+        }),
+      });
+
+      const data = await response.json();
+
+      if (response.ok) {
+        console.log("BMI entry successfully saved:", data);
+      } else {
+        console.log("Failed to save BMI:", data.message);
+      }
+    } catch (error) {
+      console.error("Error saving BMI:", error);
+    }
+  };
+
+  // Calculate progress toward the target BMI and round to 2 decimal places
+  const calculateBmiProgress = (bmiValue) => {
+    if (bmiValue && targetBmi) {
+      return Math.min((bmiValue / targetBmi) * 100, 100).toFixed(2);
+    }
+    return 0;
+  };
+
+  return (
+    <div className="mainWrapper">
+      <div className="container1">
+        <div className="titleContainer">BMI Input</div>
+        <input
+          type="text"
+          placeholder="Enter your weight (kg)"
```

Comment:
is this expecting the value has to be in kg? or what if im american and only know lbs?